query/src/storages/system/engines_table.rs

```diff
@@ -0,0 +1,90 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::any::Any;
+use std::sync::Arc;
+
+use common_datablocks::DataBlock;
+use common_datavalues::prelude::*;
+use common_exception::Result;
+use common_meta_types::TableIdent;
+use common_meta_types::TableInfo;
+use common_meta_types::TableMeta;
+use common_planners::ReadDataSourcePlan;
+use common_streams::DataBlockStream;
+use common_streams::SendableDataBlockStream;
+
+use crate::sessions::QueryContext;
+use crate::storages::Table;
+
+pub struct EnginesTable {
+    table_info: TableInfo,
+}
+
+impl EnginesTable {
+    pub fn create(table_id: u64) -> Self {
+        let schema = DataSchemaRefExt::create(vec![
+            DataField::new("Engine", DataType::String, false),
+            DataField::new("Support", DataType::String, false),
+            DataField::new("Comment", DataType::String, false),
+        ]);
+
+        let table_info = TableInfo {
+            desc: "'system'.'engines'".to_string(),
+            name: "engines".to_string(),
+            ident: TableIdent::new(table_id, 0),
+            meta: TableMeta {
+                schema,
+                engine: "SystemEngines".to_string(),
+                ..Default::default()
+            },
+        };
+        EnginesTable { table_info }
+    }
+}
+
+#[async_trait::async_trait]
+impl Table for EnginesTable {
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn get_table_info(&self) -> &TableInfo {
+        &self.table_info
+    }
+
+    async fn read(
+        &self,
+        _ctx: Arc<QueryContext>,
+        _plan: &ReadDataSourcePlan,
+    ) -> Result<SendableDataBlockStream> {
+        let mut engine_name = Vec::with_capacity(1);
+        let mut engine_support = Vec::with_capacity(1);
+        let mut engine_comment = Vec::with_capacity(1);
+        // fuse engine
+        engine_name.push("FUSE".as_bytes());
```

Comment:
Let's get the engines information from the StorageFactory :\

Reply:
https://github.com/datafuselabs/databend/blob/76e831b753eb85b274314e3ec8c4a30c0be48171/query/src/storages/storage_factory.rs#L51-L73

Reply:
ok, let me fix it

Reply:
Looks like we should add `desc()` to `StorageCreator` as the `comment` column:
```
pub trait StorageCreator: Send + Sync {
    fn try_create(&self, ctx: StorageContext, table_info: TableInfo) -> Result<Box<dyn Table>>;
    fn desc(&self) -> &str;
}
```